server/src/main/kotlin/suwayomi/tachidesk/server/ServerConfig.kt

```diff
@@ -8,58 +8,124 @@ package suwayomi.tachidesk.server
+    val downloadsPath: MutableStateFlow<String> by OverrideConfigValue(StringConfigAdapter)
+    val autoDownloadNewChapters: MutableStateFlow<Boolean> by OverrideConfigValue(BooleanConfigAdapter)
+
+    // requests
+    val maxSourcesInParallel: MutableStateFlow<Int> by OverrideConfigValue(IntConfigAdapter)
 
     // updater
-    var maxParallelUpdateRequests: Int by overridableConfig
-    var excludeUnreadChapters: Boolean by overridableConfig
-    var excludeNotStarted: Boolean by overridableConfig
-    var excludeCompleted: Boolean by overridableConfig
-    var globalUpdateInterval: Double by overridableConfig
+    val excludeUnreadChapters: MutableStateFlow<Boolean> by OverrideConfigValue(BooleanConfigAdapter)
+    val excludeNotStarted: MutableStateFlow<Boolean> by OverrideConfigValue(BooleanConfigAdapter)
+    val excludeCompleted: MutableStateFlow<Boolean> by OverrideConfigValue(BooleanConfigAdapter)
+    val globalUpdateInterval: MutableStateFlow<Double> by OverrideConfigValue(DoubleConfigAdapter)
 
     // Authentication
-    var basicAuthEnabled: Boolean by overridableConfig
-    var basicAuthUsername: String by overridableConfig
-    var basicAuthPassword: String by overridableConfig
+    val basicAuthEnabled: MutableStateFlow<Boolean> by OverrideConfigValue(BooleanConfigAdapter)
+    val basicAuthUsername: MutableStateFlow<String> by OverrideConfigValue(StringConfigAdapter)
+    val basicAuthPassword: MutableStateFlow<String> by OverrideConfigValue(StringConfigAdapter)
 
     // misc
-    var debugLogsEnabled: Boolean = debugLogsEnabled(GlobalConfigManager.config)
-    var systemTrayEnabled: Boolean by overridableConfig
+    val debugLogsEnabled: MutableStateFlow<Boolean> by OverrideConfigValue(BooleanConfigAdapter)
+    val systemTrayEnabled: MutableStateFlow<Boolean> by OverrideConfigValue(BooleanConfigAdapter)
 
     // backup
-    var backupPath: String by overridableConfig
-    var backupTime: String by overridableConfig
-    var backupInterval: Int by overridableConfig
-    var backupTTL: Int by overridableConfig
+    val backupPath: MutableStateFlow<String> by OverrideConfigValue(StringConfigAdapter)
+    val backupTime: MutableStateFlow<String> by OverrideConfigValue(StringConfigAdapter)
+    val backupInterval: MutableStateFlow<Int> by OverrideConfigValue(IntConfigAdapter)
+    val backupTTL: MutableStateFlow<Int> by OverrideConfigValue(IntConfigAdapter)
 
     // local source
-    var localSourcePath: String by overridableConfig
+    val localSourcePath: MutableStateFlow<String> by OverrideConfigValue(StringConfigAdapter)
+
+    fun <T> subscribeTo(flow: Flow<T>, onChange: suspend (value: T) -> Unit, ignoreInitialValue: Boolean = true) {
+        val actualFlow = if (ignoreInitialValue) {
+            flow.drop(1)
+        } else {
+            flow
+        }
+
+        actualFlow.distinctUntilChanged().onEach { onChange(it) }.conflate().launchIn(mutableConfigValueScope)
```

Comment:
I don't think you need to Conflate here

Reply:
I blame ChatGPT for giving me an incorrect solution (and definitely not me for not correctly checking if that is working 😅)

the idea is to have the latest value wait until the current handling of the last emitted value to be finished and omit all values in between, so basically this https://github.com/Suwayomi/Tachidesk-Server/pull/639/files#diff-65dbed49da02019a15cea5b7d68041ebd4b65a0b5bee5daf91fba2e47b07401bR146

Reply:
fixed

Reply:
Hmm now that I think about it, couldn't you just use a Mutex with Mutex.tryLock instead of withLock?

Reply:
Actually ignore that, it may break something